diff --git a/dbus-1.pcin b/dbus-1.pcin
index cac34fb..298c00c 100644
--- a/dbus-1.pcin
+++ b/dbus-1.pcin
@@ -10,6 +10,6 @@ daemondir=/tmp/kai/bin
 
 Name: dbus
 Description: D-bus replacement stub
-Version: 1.4.20
+Version: 1.5
 Libs: -L${libdir} -lnobus
 Cflags: -I${includedir}/dbus-1.0 -I${libdir}/dbus-1.0/include
diff --git a/include/dbus-1.0/dbus/dbus-message.h b/include/dbus-1.0/dbus/dbus-message.h
index fd36a95..cee76f1 100644
--- a/include/dbus-1.0/dbus/dbus-message.h
+++ b/include/dbus-1.0/dbus/dbus-message.h
@@ -95,7 +95,7 @@ DBusMessage* dbus_message_copy (const DBusMessage *message);
 
 DBusMessage* dbus_message_ref (DBusMessage *message);
 
-#define dbus_message_unref(a) ((void)0)
+DBusMessage* dbus_message_unref (DBusMessage *message);
 
 int dbus_message_get_type (DBusMessage *message);
 
diff --git a/stubs.c b/stubs.c
index 39c02b5..56652fa 100644
--- a/stubs.c
+++ b/stubs.c
@@ -707,6 +707,10 @@ DBusMessage* dbus_message_ref (DBusMessage *message) {
 	return message;
 }
 
+DBusMessage* dbus_message_unref (DBusMessage *message) {
+    return NULL;
+}
+
 int dbus_message_get_type (DBusMessage *message) {
 	return 0;
 }
