diff --git a/src/config.def.h b/src/config.def.h
index fcdcc03..c9437f1 100644
--- a/src/config.def.h
+++ b/src/config.def.h
@@ -1,4 +1,4 @@
-#define TTY_COLOR_HIGHLIGHT TTY_COLOR_YELLOW
+#define TTY_COLOR_HIGHLIGHT TTY_COLOR_BLUE
 
 #define SCORE_GAP_LEADING -0.005
 #define SCORE_GAP_TRAILING -0.005
diff --git a/src/tty_interface.c b/src/tty_interface.c
index 343dde8..8322d16 100644
--- a/src/tty_interface.c
+++ b/src/tty_interface.c
@@ -67,6 +67,9 @@ static void draw_match(tty_interface_t *state, const char *choice, int selected)
 		}
 		if (choice[i] == '\n') {
 			tty_putc(tty, ' ');
+		} else if (choice[i] == '\t') {
+			tty_putc(tty, ' ');
+			tty_putc(tty, ' ');
 		} else {
 			tty_printf(tty, "%c", choice[i]);
 		}
