diff --git a/lib/contrib/gnome-languages.c b/lib/contrib/gnome-languages.c
index 75ba956..04aa8cd 100644
--- a/lib/contrib/gnome-languages.c
+++ b/lib/contrib/gnome-languages.c
@@ -44,8 +44,6 @@
 #define __LC_LAST       13
 #endif
 
-#define ISO_CODES_DATADIR ISO_CODES_PREFIX "/share/xml/iso-codes"
-#define ISO_CODES_LOCALESDIR ISO_CODES_PREFIX "/share/locale"
 
 #if 0
 #include "default-input-sources.h"
@@ -970,40 +968,7 @@ territories_parse_start_tag (GMarkupParseContext      *ctx,
 static void
 languages_variant_init (const char *variant)
 {
-        gboolean res;
-        gsize    buf_len;
-        g_autofree char *buf = NULL;
-        g_autofree char *filename = NULL;
-        g_autoptr (GError) error = NULL;
-
-        bindtextdomain (variant, ISO_CODES_LOCALESDIR);
-        bind_textdomain_codeset (variant, "UTF-8");
-
-        error = NULL;
-        filename = g_strdup_printf (ISO_CODES_DATADIR "/%s.xml", variant);
-        res = g_file_get_contents (filename,
-                                   &buf,
-                                   &buf_len,
-                                   &error);
-        if (res) {
-                g_autoptr (GMarkupParseContext) ctx = NULL;
-                GMarkupParser        parser = { languages_parse_start_tag, NULL, NULL, NULL, NULL };
-
-                ctx = g_markup_parse_context_new (&parser, 0, NULL, NULL);
-
-                error = NULL;
-                res = g_markup_parse_context_parse (ctx, buf, buf_len, &error);
-
-                if (! res) {
-                        g_warning ("Failed to parse '%s': %s\n",
-                                   filename,
-                                   error->message);
-                }
-        } else {
-                g_warning ("Failed to load '%s': %s\n",
-                           filename,
-                           error->message);
-        }
+
 }
 
 static void
@@ -1021,43 +986,7 @@ languages_init (void)
 static void
 territories_init (void)
 {
-        gboolean res;
-        gsize    buf_len;
-        g_autofree char *buf = NULL;
-        g_autoptr (GError) error = NULL;
-
-        if (gnome_territories_map)
-                return;
-
-        bindtextdomain ("iso_3166", ISO_CODES_LOCALESDIR);
-        bind_textdomain_codeset ("iso_3166", "UTF-8");
 
-        gnome_territories_map = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_free);
-
-        error = NULL;
-        res = g_file_get_contents (ISO_CODES_DATADIR "/iso_3166.xml",
-                                   &buf,
-                                   &buf_len,
-                                   &error);
-        if (res) {
-                g_autoptr (GMarkupParseContext) ctx = NULL;
-                GMarkupParser        parser = { territories_parse_start_tag, NULL, NULL, NULL, NULL };
-
-                ctx = g_markup_parse_context_new (&parser, 0, NULL, NULL);
-
-                error = NULL;
-                res = g_markup_parse_context_parse (ctx, buf, buf_len, &error);
-
-                if (! res) {
-                        g_warning ("Failed to parse '%s': %s\n",
-                                   ISO_CODES_DATADIR "/iso_3166.xml",
-                                   error->message);
-                }
-        } else {
-                g_warning ("Failed to load '%s': %s\n",
-                           ISO_CODES_DATADIR "/iso_3166.xml",
-                           error->message);
-        }
 }
 
 /**
diff --git a/meson.build b/meson.build
index ab00889..f720bd0 100644
--- a/meson.build
+++ b/meson.build
@@ -28,9 +28,6 @@ pkglibexecdir = join_paths(libexecdir, 'epiphany')
 servicedir = join_paths(datadir, 'dbus-1', 'services')
 webprocessextensionsdir = join_paths(pkglibdir, 'web-process-extensions')
 
-iso_codes = dependency('iso-codes')
-iso_codes_prefix = iso_codes.get_pkgconfig_variable('prefix')
-
 profile = get_option('profile')
 application_id = 'org.gnome.Epiphany'
 
@@ -46,7 +43,6 @@ conf.set10('DEVELOPER_MODE', developer_mode)
 conf.set_quoted('DISTRIBUTOR_NAME', distributor_name)
 conf.set_quoted('EPHY_WEB_PROCESS_EXTENSIONS_DIR', webprocessextensionsdir)
 conf.set_quoted('GETTEXT_PACKAGE', meson.project_name())
-conf.set_quoted('ISO_CODES_PREFIX', iso_codes_prefix)
 conf.set_quoted('LIBLOCALEDIR', liblocaledir)
 conf.set_quoted('LOCALEDIR', localedir)
 conf.set_quoted('PKGLIBEXECDIR', pkglibexecdir)
@@ -88,7 +84,6 @@ gsettings_desktop_schemas = dependency('gsettings-desktop-schemas')
 gtk_dep = dependency('gtk+-3.0', version: gtk_requirement)
 gtk_unix_print_dep = dependency('gtk+-unix-print-3.0', version: gtk_requirement)
 hogweed_dep = dependency('hogweed', version: nettle_requirement)
-iso_codes_dep = dependency('iso-codes', version: '>= 0.35')
 json_glib_dep = dependency('json-glib-1.0', version: '>= 1.2.4')
 libdazzle_dep = dependency('libdazzle-1.0', version: '>= 3.37.1')
 libhandy_dep = dependency('libhandy-1', version: '>= 1.0.0')
